### DISCUSSION TITLE
relocate appsettings.json of global tenant outsite the original path App_Data to path inside the project Project.web/NewFIle

### DISCUSSION THREAD

**Discussion by ODA669** [OrchardCMS/OrchardCore#15241] [General]
Good morning, can someone help me. I need to place appsettings,json in global tenants outside of this app_data/sites/default/ path and place it in proiject.web/default/ to place it in this appsettings,json and appsettings.production.json folder. how do I do this thank you

**Comment by Piedone** [comment]
Please check out https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/.

**Reply by ODA669** [reply]
Thanks for your answer, I created a tenant and when I excluded it it worked, but when appsetting.json that this app_data I place it in a folder inside the project it does not work in this application configuration, it is the connection string to the database where it is created orchardcore, and I look in the orchard codre shell module and I can't find the path for when the system reads that

original path  --------------------- orchardcore project/App_Data/Sites/Default/appsettings.json
the path i want -------------------orchardcore project/Conecction/appsetting.jason

**Reply by Piedone** [reply]
I see. You'd need to override a lot of things in OC to make that possible.

However, can you explain why do you want to do this? On the linked docs page you can see how to apply the same tenant configuration from a root appsettings file in your web project ([see here](https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/#tenant-preconfiguration), as well as from any configuration provider ASP.NET Core supports. This is a lot of flexibility already, isn't it suitable for your use case?

**Reply by ODA669** [reply]
i need to do this because when i do publish in production It damages the files in production and overwrites them and I need it to be one for production and one for development but that it does not publish the appsetting only production and that it works, if I can use it outside of the app file
_data I can exclude it and not publish it, I managed to do it with other files but precisely with this json I have problems, I don't know where the system determines that it should be in app_data that should read this json 

this link don't have the information that  i need
https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/#tenant-preconfiguration

**Reply by Piedone** [reply]
You then either need to publish in a way that it doesn't delete the file, or use a different shell configuration provider so there won't be a file: https://docs.orchardcore.net/en/latest/docs/reference/core/Shells/.

**Reply by ODA669** [reply]
i did this, but the probleme don't its the plublish, the probleme are that appsettings.json i can't  I can't move it to another folder outside of app_data
 how ican change the provider for other file ex: appsettings.pro.json
![image](https://github.com/OrchardCMS/OrchardCore/assets/60682935/23f86d73-8def-4f3c-bfe8-8262524b060f)

**Reply by Piedone** [reply]
That's not a problem but a potential solution, andI think it's a wrong one :). I suggest doing what I've written above.

**Comment by ns8482e** [comment]
Did you checkout the documentation link above? Checkout section `ORCHARD_APP_DATA` Environment Variable.  That will  cover all your needs to move out the App_Data from application folder

**Reply by ODA669** [reply]
If I have read the link but it did not work, could you tell me how to change it?
how ican change the provider path for other file ex: c://newFile/appsettings.pro.json
![Capture d’écran 2024-02-08 101543](https://github.com/OrchardCMS/OrchardCore/assets/60682935/8d6fc937-2a10-47cb-91f4-9e4580df6855)

**Reply by ns8482e** [reply]
You can provide environment variable in your launch settings. See how I am using. This will create `tenant.json` `sites`folder and each tenant under `C:\oc_data`

![image](https://github.com/OrchardCMS/OrchardCore/assets/23270244/c5909fb0-062e-4b50-bcdc-bf9ac8667cab)

**Reply by ODA669** [reply]
thanks so much for your help, this is what i looking for!!

**Reply by ODA669** [reply]
you can say me how works in production when i publied the project in visual studio, it search the old path, how i should that it seach the new path inside the envirenement variable? in local it work but when i publied  take the old path @Piedone @sebastienros   , maybe i need to add any aditional configuration in pubxml?

**Reply by ns8482e** [reply]
you have to set environment variable on server!

**Reply by ODA669** [reply]
how i do that, in pubxml i can't find info of how do this, can you explaine to me?

**Reply by ns8482e** [reply]
Where are you deploying IIS?

**Reply by ODA669** [reply]
yes

**Reply by ns8482e** [reply]
On your server, Open the Control Panel > System > Advanced system settings and add environment variable.  See detailed Microsoft's doc here https://learn.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-8.0

And here https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/web-config?view=aspnetcore-8.0#set-environment-variables

**Reply by ODA669** [reply]
i will try, thanks
